python/_tskitmodule.c

```diff
@@ -6303,6 +6585,32 @@ TreeSequence_get_site(TreeSequence *self, PyObject *args)
     return ret;
 }
 
+static PyObject *
+TreeSequence_get_metadata_schemas(TreeSequence *self) {
+    PyObject *ret = NULL;
+    tsk_table_collection_t *tables = self->tree_sequence->tables;
+    ret = PyStructSequence_New(&MetadataSchemas);
+    if (ret == NULL) {
+        goto out;
+    }
+    PyStructSequence_SetItem(ret, 0, PyBytes_FromStringAndSize(
```

Comment:
These can fail, right? I think we'd need to check every one of these calls and clean up appropriately on errors.

I know it's tedious, but what about ``TreeSequence_get_node_metadata_schema()``, TreeSequence_get_individual_metadata_schema()``, etc? It's a bunch of boilerplate, but it's simple and easy to write in a bulletproof way.

Reply:
Fixed in https://github.com/tskit-dev/tskit/pull/491/commits/338a045e3f6586e4673076b4e2fd62ef05e9ef7f I've added low-level tests in https://github.com/tskit-dev/tskit/pull/491/commits/e3f97bd90d3a0eb95dca5a2098c19f7b27174e99